wal/decoder.go

```diff
@@ -98,6 +113,44 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	return nil
 }
 
+// isTornEntry determines whether the last entry of the WAL was partially written
+// and corrupted because of a torn write.
+func (d *decoder) isTornEntry(data []byte) bool {
```

Comment:
Sorry-- A is a single record that spans three sectors but has a write tear in the middle. B is a record that follows A. The brackets indicate sector boundaries.